.github/workflows/lint.yml

```diff
@@ -0,0 +1,37 @@
+name: Linting
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6']
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Get Pip cache directory
+      id: pip-cache
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+    - uses: actions/cache@v1
+      id: cache
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: ${{ runner.os }}-pip-cache
+        restore-keys: |
+          ${{ runner.os }}-pip-cache
+    - name: Install dependencies
+      run: |
+        pip install --upgrade mypy black
+    - name: Lint
+      run: |
+        black --check .
```

Comment:
Super excited to have black! How does it work, does this validate that black was run by the user, and fail otherwise? 

I have no idea how these workflows typically work. Do people ever autoformat in the cloud?

Reply:
Yeah, I haven't seen anyone do automatic formatting in CI, so this will fail on style issues like you're saying (because of the `--check` flag). The cool thing is that black can format entire projects in one command so it's very easy to fix any issues it finds!

Reply:
btw, you can add black as an external tool to PyCharm:
https://black.readthedocs.io/en/stable/editor_integration.html

Reply:
I do that! It's really nice. I can just bang out some ugly code like an absolute maniac and Black will fix it.